pkg/controllermanager/controller/managedseedset/managedseedset_reconciler.go

```diff
@@ -0,0 +1,161 @@
+			err = fmt.Errorf("could not update status: %w", updateErr)
+		}
+	}()
+
+	// Reconcile creation or update
+	r.getLogger(set).Debug("Reconciling creation or update")
+	if status, err = r.actuator.Reconcile(ctx, set); err != nil {
+		r.errorEventf(set, "Could not reconcile creation or update: %+v", err)
+		return reconcile.Result{}, fmt.Errorf("could not reconcile ManagedSeedSet %s creation or update: %w", kutil.ObjectName(set), err)
+	}
+	r.getLogger(set).Debug("Creation or update reconciled")
+
+	// Return success result
+	return reconcile.Result{RequeueAfter: r.cfg.SyncPeriod.Duration}, nil
+}
+
+func (r *reconciler) delete(ctx context.Context, set *seedmanagementv1alpha1.ManagedSeedSet) (result reconcile.Result, err error) {
+	// Check gardener finalizer
+	if !controllerutil.ContainsFinalizer(set, gardencorev1beta1.GardenerName) {
+		r.getLogger(set).Debug("Skipping as it does not have a finalizer")
+		return reconcile.Result{}, nil
+	}
+
+	var status *seedmanagementv1alpha1.ManagedSeedSetStatus
+	defer func() {
+		// Update status, on failure return the update error unless there is another error
+		if updateErr := r.updateStatus(ctx, set, status); updateErr != nil && err == nil {
+			err = fmt.Errorf("could not update status: %w", updateErr)
+		}
+	}()
+
+	// Reconcile deletion
+	r.getLogger(set).Debug("Reconciling deletion")
+	if status, err = r.actuator.Reconcile(ctx, set); err != nil {
+		r.errorEventf(set, "Could not reconcile deletion: %+v", err)
+		return reconcile.Result{}, fmt.Errorf("could not reconcile ManagedSeedSet %s deletion: %w", kutil.ObjectName(set), err)
+	}
+	r.getLogger(set).Debug("Deletion reconciled")
+
+	// If there is no pending replica, remove gardener finalizer
+	if status.PendingReplica == nil {
+		r.getLogger(set).Debug("Removing finalizer")
+		if err := controllerutils.PatchRemoveFinalizers(ctx, r.gardenClient.Client(), set, gardencorev1beta1.GardenerName); err != nil {
+			return reconcile.Result{}, fmt.Errorf("could not remove gardener finalizer: %w", err)
+		}
+		return reconcile.Result{}, nil
+	}
+
+	// Return success result
+	return reconcile.Result{RequeueAfter: r.cfg.SyncPeriod.Duration}, nil
+}
+
+func (r *reconciler) errorEventf(set *seedmanagementv1alpha1.ManagedSeedSet, fmt string, args ...interface{}) {
+	r.recorder.Eventf(set, corev1.EventTypeWarning, gardencorev1beta1.EventReconcileError, fmt, args...)
+	r.getLogger(set).Errorf(fmt, args...)
+}
+
+func (r *reconciler) getLogger(set *seedmanagementv1alpha1.ManagedSeedSet) *logrus.Entry {
+	return logger.NewFieldLogger(r.logger, "managedSeedSet", kutil.ObjectName(set))
+}
+
+func (r *reconciler) updateStatus(ctx context.Context, set *seedmanagementv1alpha1.ManagedSeedSet, status *seedmanagementv1alpha1.ManagedSeedSetStatus) error {
+	return kutil.TryPatchStatus(ctx, retry.DefaultBackoff, r.gardenClient.Client(), set, func() error {
+		if status != nil {
+			set.Status = *status
+		}
+		return nil
+	})
```

Comment:
`TryPatchStatus` patches the status with optimistic locking, so it can likely lead to conflicts (especially with cached clients).
We should instead use a strategic merge patch without optimistic locking here:
```suggestion
if status == nil {
		return nil
	}
	patch := client.StrategicMergeFrom(set.DeepCopy())
	set.Status = *status
	return r.gardenClient.Client().Status().Patch(ctx, set, patch)
```

Reply:
OK

Reply:
Done